### DISCUSSION TITLE
[Application Development] <Does the SuperchainERC20 will compatible existing ERC20 on L2>

### DISCUSSION THREAD

**Discussion by zhwindy** [ethereum-optimism/developers#660] [Application Development]
### Did you check the documentation?

- [X] Yes
- [ ] No

### What type of issue are you experiencing?

- [ ] Error
- [ ] Feature Request
- [X] General Question

### Issue Description

I'm reading the Docs of the OP Interoperability and SuperchainERC20, and I'm curious about does the new SuperchainERC20 will compatible existing ERC20 on L2? The documentation doesn't mention this.

### Steps to Reproduce

no

### Are you using a specific library or SDK?

- [ ] OP-SDK
- [ ] Viem
- [X] Other
- [ ] Not Sure

### Are you using the latest version of the library/SDK?

- [X] Yes
- [ ] No

### What type of support do you need?

- [X] Technical support for issue resolution
- [ ] Feature implementation guidance
- [ ] General advice or consultation
- [X] Documentation or learning resources

**Comment by hash1go** [comment]
Hello!
SuperchainERC20 covers the functionality of ERC20.
https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/SuperchainERC20.sol#L7

**Reply by zhwindy** [reply]
yep, I understand this. My question is dose it possible to Migrate L2 exists ERC20 onchain Token to a SuperchainERC20 Token? such as Wrap or something else.

**Reply by hash1go** [reply]
Migrating from ERC20 to SuperchainERC20 presents the same issues as migrating any standard ERC20 contract.
Therefore, when application builders migrate their existing ERC20 to SuperchainERC20, they will need to communicate well with their community and complete the migration process carefully.

**Reply by zhwindy** [reply]
Actually what I want to know is whether it is feasible, whether it is technically feasible, or whether there will be feasible solutions in the future. Rather than considering the community's problems because there is no actual migration.

**Reply by hash1go** [reply]
It is possible to migrate by implementing functionality to burn existing ERC20s and mint SuperchainERC20s.
According to my understanding, contract migration cannot be done without ERC20 holders being aware of the migration process. This is because some tokens are locked in DeFi contracts.
